File: Python/balancedParens.py

def balancedParens(string):
  stack = []
  pairs = {
    '(':')',
    '[':']',
    '{':'}'
  }

  for i in string:
    if i in pairs:
      stack.append(pairs[i]);
    elif i == ')' or i == '}' or i == ']':
      if stack.pop() != i:
        return False


  print len(stack) == 0

balancedParens('(){}[')

File: JS/balancedParens.js
var balancedParens = function(str) {
  var stack = [];
  var pairs = {
    '(':')',
    '[':']',
    '{':'}'
  };
  for (var i = 0; i< str.length; i++) {
    if (pairs[str[i]]) {
      stack.push(pairs[str[i]]);
    } else if (str[i] === ')' || str[i] === ']' || str[i] === '}') {
      if (stack.pop() !== str[i]) {
        return false;
      }
    }
  }
  return stack.length === 0;
};

